File: app/src/main/java/ru/fylmr/diploma/crypto/primitives/curve25519/ge_p3_0.java
package ru.fylmr.diploma.crypto.primitives.curve25519;

public class ge_p3_0 {

//CONVERT #include "ge.h"

    public static void ge_p3_0(GroupElemExtended h) {
        new fe_0(h.X);
        new fe_1(h.Y);
        new fe_1(h.Z);
        new fe_0(h.T);
    }


}

File: app/src/main/java/ru/fylmr/diploma/crypto/primitives/curve25519/ge_p3_0.kt
package ru.fylmr.diploma.crypto.primitives.curve25519

/**
 * ge_p3_0
 */
fun geExtendedZero(h: GEExtended) {
    //CONVERT #include "ge.h"
    feZero(h.X.bytes)
    feOne(h.Y.bytes)
    feOne(h.Z.bytes)
    feZero(h.T.bytes)
}
